R/networks.R

```diff
@@ -43,13 +43,21 @@ gce_make_network <- function(name,
   if(!is.null(externalIP)){
```

Comment:
The logic is a bit confusing, in summary should it be the below?

* If `externalIP` is NULL make an `ac` with natIP = NULL
* If `externalIP` is "none" then make `ac` NULL
* If `externalIP` is an IP then use that within `ac`

so perhaps a little utility function to keep the code clear:

```r
make_ac <- function(externalIP, name){
  if(!is.null(externalIP) && externalIP == "none") return(NULL)

  ## externalIP can be NULL
  list(
    natIP = jsonlite::unbox(externalIP),
    type = jsonlite::unbox("ONE_TO_ONE_NAT"),
    name = jsonlite::unbox(name)
  )
}
```

By default all list elements that are `NULL`get removed from the body calls within `googleAuthR`

Reply:
```r
> make_ac(NULL, "hi")
$natIP
NULL

$type
[x] "ONE_TO_ONE_NAT"

$name
[x] "hi"

> make_ac("none", "hi")
NULL
> make_ac("123.345.345.343", "hi")
$natIP
[x] "123.345.345.343"

$type
[x] "ONE_TO_ONE_NAT"

$name
[x] "hi"
```

Reply:
Absolutely. That is a lot clearer! Let me go ahead and make the changes and test it out.

Reply:
Done.